airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/AbstractPostgresSourceDatatypeTest.java

```diff
@@ -0,0 +1,526 @@
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("cidr")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("null", "'192.168.100.128/25'", "'192.168/24'", "'192.168.1'",
+                "'128.1'", "'2001:4f8:3:ba::/64'")
+            .addExpectedValues(null, "192.168.100.128/25", "192.168.0.0/24", "192.168.1.0/24",
+                "128.1.0.0/16", "2001:4f8:3:ba::/64")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("circle")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("'(5,7),10'", "'(0,0),0'", "'(-10,-4),10'", "null")
+            .addExpectedValues("<(5.0,7.0),10.0>", "<(0.0,0.0),0.0>", "<(-10.0,-4.0),10.0>", null)
+            .build());
+
+    // Debezium does not handle era indicators (AD nd BC)
+    // https://github.com/airbytehq/airbyte/issues/14590
+    // addDataTypeTestData(
+    // TestDataHolder.builder()
+    // .sourceType("date")
+    // .airbyteType(JsonSchemaType.STRING_DATE)
+    // .addInsertValues("'1999-01-08'", "'1991-02-10 BC'", "null")
+    // .addExpectedValues("1999-01-08", "1990-02-10 BC", null)
+    // .build());
+
+    for (final String type : Set.of("double precision", "float", "float8")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .airbyteType(JsonSchemaType.NUMBER)
+              .addInsertValues("'123'", "'1234567890.1234567'", "null")
+              // Postgres source does not support these special values yet
+              // https://github.com/airbytehq/airbyte/issues/8902
+              // "'-Infinity'", "'Infinity'", "'NaN'", "null")
+              .addExpectedValues("123.0", "1.2345678901234567E9", null)
+              // "-Infinity", "Infinity", "NaN", null)
+              .build());
+    }
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("inet")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("'198.24.10.0/24'", "'198.24.10.0'", "'198.10/8'", "null")
+            .addExpectedValues("198.24.10.0/24", "198.24.10.0", "198.10.0.0/8", null)
+            .build());
+
+    for (final String type : Set.of("integer", "int", "int4")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .airbyteType(JsonSchemaType.INTEGER)
+              .addInsertValues("null", "1001", "-2147483648", "2147483647")
+              .addExpectedValues(null, "1001", "-2147483648", "2147483647")
+              .build());
+    }
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("interval")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("null", "'P1Y2M3DT4H5M6S'", "'-178000000'", "'178000000'")
+            .addExpectedValues(null, "1 year 2 mons 3 days 04:05:06", "-49444:26:40", "49444:26:40")
```

Comment:
is it expected that the input format also affects the output format (i.e. `HHHHHHHHH:MM:SS` vs `x years y mons etc`)